File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

//initialize variables for worth of each coin along with other variables
int q = 25;
int d = 10;
int n = 5;
int p = 1;
float dollars;
int cents;
int coins;

int main(void)
{
//do loop to check to make sure that the user enters in a non-negative number using provided get_float function
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

//round function to change float to an integer
    cents = round(dollars * 100);
    coins = 0;

//series of while loops that try to divide cents by coin value, iterate coin count, subtract that value from cents
    while ((cents / q) >= 1)
    {
        coins++;
        cents = cents - q;
    }

    while ((cents / d) >= 1)
    {
        coins++;
        cents = cents - d;
    }

    while ((cents / n) >= 1)
    {
        coins++;
        cents = cents - n;
    }

    while ((cents / p) >= 1)
    {
        coins++;
        cents = cents - p;
    }

//print out final answer
    printf("%i\n", coins);


}

File: pset6/cash/cash.py
# import get float function from cs50 library
from cs50 import get_float

# initialize variables for each coin value
q = 25
d = 10
n = 5
p = 1

# ask user for float until it is positive
while True:
    dollars = get_float("Change owed: ")
    if dollars > 0:
        break

# convert and round dollars to cents
cents = round(dollars * 100)

# initialize a variable to keep track of how many times each coin is used in algorithm
coins = 0

# series of while loops that try to divide cents by coin value, iterate coin count, subtract that value from cents
while (cents / q) >= 1:
    coins += 1
    cents -= q

while (cents / d) >= 1:
    coins += 1
    cents -= d

while (cents / n) >= 1:
    coins += 1
    cents -= n

while (cents / p) >= 1:
    coins += 1
    cents -= p

print(coins)